core/new-gui/src/app/common/service/user/workflow-persist/workflow-persist.service.ts

```diff
@@ -15,21 +16,26 @@ export class WorkflowPersistService {
   constructor(public http: HttpClient) {
   }
 
-  public saveWorkflow(savedWorkflow: string, workflowName: string, workflowID: string | null): Observable<Workflow> {
+  public persistWorkflow(cachedWorkflow: Workflow, workflowName: string, workflowID: number | null): Observable<Workflow> {
```

Comment:
seems like this service's logic and functionaility is simple, we also have another `cacheWorkflowService` , it's confusing to have two services that do handle similar things without clear documentations on the difference. 

I suggest we merge it with `CacheWorkflowService`, just have one single service managing all related logic

Reply:
No, I designed it to be separate. The logic is as follows:

1. the `CacheWorkflowService` should handle frontend cache (in `localStorage` without any HTTP request to the backend. The logic should happen purely in the frontend.
2. the `workflowPersistService` would handle backend persisting in the database. Right now it seems simple and shallow,  but later it will be extended with version controlling on the workflow.

Those two logics should be parallel and independent of each other. So I am very against the idea of merging them.